packages/keystatic/src/component-blocks/cloud-image-preview.tsx

```diff
@@ -10,65 +10,189 @@ import { Icon } from '@keystar/ui/icon';
     try {
-      const data: unknown = JSON.parse(text);
+      const data: any = JSON.parse(text);
       if (
         typeof data === 'object' &&
         data !== null &&
         'src' in data &&
         typeof data.src === 'string'
       ) {
-        event.preventDefault();
-        props.onChange({
-          src: data.src,
-          alt: 'alt' in data && typeof data.alt === 'string' ? data.alt : '',
-          height: 'height' in data ? getDimension(data.height) : '',
-          width: 'width' in data ? getDimension(data.width) : '',
-        });
+        setPastedData(cleanImageData(data));
         return;
       }
     } catch (err) {}
     const pattern = /^\s*!\[(.*)\]\(([a-z0-9_\-/:.]+)\)\s*$/;
     const match = text.match(pattern);
     if (match) {
-      event.preventDefault();
-      props.onChange({
-        src: match[2],
-        alt: match[1],
-        height: '',
-        width: '',
-      });
+      setPastedData(
+        cleanImageData({
+          src: match[2],
+          alt: match[1],
+        })
+      );
+      return;
+    }
+    setPastedData(cleanImageData({ src: text }));
+  };
+
+  useEffect(() => {
+    if (!pastedData || !pastedData.src) {
+      setStatus('');
+      return;
+    }
+    if (!isValidURL(pastedData.src)) {
+      setStatus('The pasted data is not a valid Image URL');
       return;
     }
+    const img = new Image();
+    setStatus('Checking pasted data...');
+    img.onload = () => {
+      onChange({
+        ...pastedData,
+        width: img.width.toString(),
+        height: img.height.toString(),
+      });
+    };
+    img.onerror = () => {
+      setStatus('The pasted data is not a valid Image URL');
+      console.log('Invalid Image URL pasted:', pastedData);
+    };
+    img.src = pastedData.src;
+    return () => {
+      img.onload = null;
+    };
+  }, [pastedData, onChange]);
+
+  return (
+    <NotEditable>
+      <VStack gap="regular">
+        <div style={{ position: 'relative' }}>
+          <input
+            type="text"
+            value=""
+            onFocus={() => setIsFocused(true)}
+            onBlur={() => setIsFocused(false)}
+            onClick={() => {
+              document.execCommand('paste');
+            }}
+            onPaste={onPaste}
+            style={{
+              position: 'absolute',
+              cursor: 'default',
+              fontSize: '16px',
+              border: 'none',
+              outline: isFocused ? '2px solid #30a46c' : 'none',
+              background: 'transparent',
+              padding: 0,
+              paddingLeft: '2%',
+              margin: 0,
+              opacity: 0,
+              color: 'transparent',
+              borderRadius: '6px',
+              width: '98%',
+              height: '100%',
+            }}
+          />
+          {isFocused ? (
+            <Flex
+              backgroundColor="positive"
+              border="positive"
+              color="positive"
```

Comment:
The `color` prop is only valid on typographic primitives. If you're not seeing a type error, there's a bug.

Reply:
There's a bug, I found this in the TS autocomplete so didn't think to reach for a `Text` component (also wondered why it had no effect, but figured I'd leave that for the review)

Reply:
Ugh, layout components are polymorphic with the `elementType` prop—I guess it's including the [legacy attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes)…

Reply:
Resolved in #563